### DISCUSSION TITLE
Octofarm and Bonjour

### DISCUSSION THREAD

**Discussion by kellyhamlin** [OctoFarm/OctoFarm#1058] [General]
Got Octofarm setup and running with 7 Octoprint instances today and getting used to how things work, so far I really like it.

My question is how do you handle PI's on DHCP and Bonjour to reach the hostnames? I run Octofarm in a docker instance on my debian server. I installed a package that lets me resolve those hostnames, but that doesn't appear to translate over to docker-networking. In other words, if I user octoprint3 or octoprint3.local, it doesnt see it. However from Debian and my other windows boxes, I can resolve that hostname just fine.

Is the proper course of action doing static IPs on my PIs? I was trying to avoid that if possible. Any solutions?

**Comment by NotExpectedYet** [comment]
> Got Octofarm setup and running with 7 Octoprint instances today and getting used to how things work, so far I really like it.
> 
> My question is how do you handle PI's on DHCP and Bonjour to reach the hostnames? I run Octofarm in a docker instance on my debian server. I installed a package that lets me resolve those hostnames, but that doesn't appear to translate over to docker-networking. In other words, if I user octoprint3 or octoprint3.local, it doesnt see it. However from Debian and my other windows boxes, I can resolve that hostname just fine.
> 
> Is the proper course of action doing static IPs on my PIs? I was trying to avoid that if possible. Any solutions?

What router do you have? 

Personally I use static assignments at the router level via MAC address of each Pi. That way I've a centralized place to manage it and it's not tied to device specific settings.

**Reply by kellyhamlin** [reply]
Thanks for the reply. I can easily do it in my unifi setup, was just trying to avoid adding to my static pool.

**Reply by NotExpectedYet** [reply]
Welcome. Yeah there's not many other alternatives really. 

If your router adds hostnames automatically to it's DNS registry then you could use that. It would be similar to Bonjour registering with mDNS, you'd just have to edit /etc/hosts (If I remember rightly) on each Pi so something unique.
